v23.1/full-text-search.md

```diff
@@ -0,0 +1,470 @@
+---
+title: Full-Text Search
+summary: Full-text searches using TSVECTOR and TSQUERY enable natural-language searches on documents with ranked results.
+toc: true
+docs_area: develop
+---
+
+A full-text search is used to perform natural-language searches on documents such as articles, websites, or other written formats.
+
+This page describes how to perform full-text searches using the provided [built-in functions](functions-and-operators.html#full-text-search-functions).
+
+{{site.data.alerts.callout_info}}
+Some PostgreSQL syntax and features are currently unsupported. For details, see [Unsupported features](#unsupported-features).
+{{site.data.alerts.end}}
+
+## How does full-text search work?
+
+In the PostgreSQL terminology, a *document* is a natural-language text, typically stored within a single database row or concatenated from multiple fields, that you [convert to a data type](#process-a-document) that is searchable using [specially formatted queries](#form-a-query).
+
+A full-text search has the following advantages over pattern matching with `LIKE` and `ILIKE`:
+
+- A full-text search can specify a [text search configuration](#text-search-configuration) that enables language-specific searches.
+- The results of a full-text search can be [ranked](#rank-search-results).
+- A full-text search can be accelerated using a [full-text index](#full-text-indexes).
+
+{{site.data.alerts.callout_success}}
+{% include {{ page.version.version }}/sql/use-case-trigram-indexes.md %}
+{{site.data.alerts.end}}
+
+### Process a document
+
+To make a document searchable, convert it to the [`TSVECTOR`](tsvector.html) data type. A `TSVECTOR` value consists of individual *lexemes*, which are normalized strings used for text matching.
+
+The `to_tsvector()` [built-in function](functions-and-operators.html#full-text-search-functions) converts a string input into a `TSVECTOR` value:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT to_tsvector('How do trees get on the internet?');
+~~~
+
+~~~
+           to_tsvector
+---------------------------------
+  'get':4 'internet':7 'tree':3
+~~~
+
+This `TSVECTOR` consists of the lexemes `get`, `internet`, and `tree`. Normalization removes the following from the input:
+
+- Suffixes. In this example, "trees" is normalized to `tree`.
+- *Stop words*. These are words that are considered not useful for indexing and searching, based on the [text search configuration](#text-search-configuration). This example does not specify a configuration, and `english` is used by default. "How", "do", "on", and "the" are identified as stop words.
+- Punctuation and capitalization.
+- Duplicates.
```

Comment:
See the other comment on the queries page. I'd include a duplicate word in the exmaple so useres can see how it works.

Reply:
Added a duplicate word to the `TSVECTOR` and `TSQUERY` syntax docs.